client/my-sites/posts/controller.js

```diff
@@ -30,6 +31,11 @@ module.exports = {
 			analyticsPageTitle = 'Blog Posts',
 			baseAnalyticsPath;
 
+		if ( !search && context.prevPath ) {
+			//Guessing search parameters from previous url
+			search = qs.parse( url.parse( context.prevPath ).query ).s;
```

Comment:
I am aware this may be considered a `Horrible ducttape hack`. The only alternative I have for this is storing search term in a Redux store, and in fact I have opened an issue:
https://github.com/Automattic/wp-calypso/issues/1112

I do not want to implement this during a 'maintenance' Milestone